qcfractal/interface/collections/dataset.py

```diff
@@ -104,6 +107,40 @@ class DataModel(Collection.DataModel):
         history: Set[Tuple[str, str, str, Optional[str], Optional[str]]] = set()
         history_keys: Tuple[str, str, str, str, str] = ("driver", "program", "method", "basis", "keywords")
 
+    def set_view(self, path: Union[str, Path]) -> None:
+        from . import HDF5View
+        self._view = HDF5View(path)
+
+    def download(self, local_path: Optional[Union[str, Path]] = None, verify: bool = True) -> None:
+        """ Download a remote view """
+        import hashlib
+
+        if self.data.view_url is None:
+            raise ValueError("A view for this dataset is not available on the server")
+
+        if local_path is not None:
+            local_path = Path(local_path)
+        else:
+            self._view_tempfile = tempfile.NamedTemporaryFile()  # keep temp file alive until self is destroyed
+            local_path = self._view_tempfile.name
+
+        r = requests.get(self.data.view_url, stream=True)
+        with open(local_path, 'wb') as fd:
+            for chunk in r.iter_content(chunk_size=8192):
+                fd.write(chunk)
+
+        if verify:
+            remote_checksum = requests.get(self.data.view_url + ".blake2b").text.rstrip()
```

Comment:
Could this be a metadata blob rather than a single file? Might help us in the future.

Reply:
I don't completely understand the proposal. Is the idea to put the checksum in the collections table? I think that would be a good idea, if that's what you mean.

Reply:
If I understand this correctly the `.blake2b` is a text file at this location. It might be more flexible for this to be a JSON-serialized dictionary so that we can have arbitrary fields.

Reply:
You understand correctly, and I'm happy to make a metadata blob instead. That said, what do you think about keeping that metadata blob in a column in the collections table rather than in a file on the http file server?

Reply:
It's more of a future proofing item. If the JSON blob just has a `blake2b` field then its fine, but what if we need something else? It would require new architecture then.

Reply:
Now a collections table column